opensrp-fp/src/main/java/org/smartregister/fp/common/repository/PreviousContactRepository.java

```diff
@@ -478,6 +479,36 @@ public Facts getImmediatePreviousSchedule(String baseEntityId, String contactNo)
         return data;
     }
 
+    public List<HashMap<String, String>> getLatestSterilizeContacts() {
+        List<HashMap<String, String>> data = new ArrayList<>();
+
+        try {
+
+            String query = "SELECT pc._id, MAX(pc.contact_no) as contact_no, pc.base_entity_id,\n " +
+                    "(SELECT ipc.value FROM previous_contact AS ipc WHERE ipc.base_entity_id = pc.base_entity_id AND ipc.key = 'sterilization_date') AS sterilize_date\n " +
+                    "FROM previous_contact AS pc\n " +
+                    "LEFT JOIN ec_client AS ec ON ec.base_entity_id = pc.base_entity_id\n " +
+                    "WHERE pc.key = 'method_exit' AND (pc.value = 'male_sterilization' OR pc.value = 'female_sterilization') AND ec.archived IS NULL\n " +
+                    "GROUP BY pc.base_entity_id";
```

Comment:
It would be nice if constants were used for table names and field names for this query.

Reply:
Done